### PR TITLE
WIP : Meross service

### PR DESCRIPTION
**Pull request by aschelch**
### Pull Request check-list

To ensure your Pull Request can be accepted as fast as possible, make sure to review and check all of these items:

- [x] If your changes affects code, did your write the tests?
- [x] Are tests passing? (`npm test` on both front/server)
- [x] Is the linter passing? (`npm run eslint` on both front/server)
- [x] Did you run prettier? (`npm run prettier` on both front/server)
- [x] If you are adding a new features/services, did you run integration comparator? (`npm run compare-translations` on front)
- [ ] If you are adding a new features/services which needs explanation, did you modify the user documentation? See [the GitHub repo](https://github.com/GladysAssistant/v4-website) and the [website](https://gladysassistant.com).
- [ ] Did you add fake requests data for the demo mode (`front/src/config/demo.json`) so that the demo website is working without a backend? (if needed) See [https://demo.gladysassistant.com](https://demo.gladysassistant.com).

NOTE: these things are not required to open a PR and can be done afterwards / while the PR is open.

### Description of change

Add a service to handle Meross plug (tested with MSS210).

Capture d’écran 2020-11-19 à 22 51 21
Capture d’écran 2020-11-19 à 22 52 33
Capture d’écran 2020-11-19 à 22 52 45

### CODE REVIEW

File: server/services/meross/utils.js [hunk 1]
```diff
@@ -0,0 +1,296 @@
+/**
+ * Generate MD5.
+ *
+ * @see http://www.webtoolkit.info/
+ * @param string - String to md5.
+ * @returns MD5 result.
+ * */
+function md5(string) {
```
Comment (cicoub13) [inline server/services/meross/utils.js]: Maybe include https://www.npmjs.com/package/jshashes or https://www.npmjs.com/package/md5 ?

Comment (aschelch) [reply server/services/meross/utils.js]: You are right 👍 
It's done ;)

File: server/services/meross/lib/plug/plug.getState.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+ * @private
+ * @description Get the current state of a device.
+ * @param {Object} device - The device object.
+ * @param {Object} deviceFeature - The deviceFeature object.
+ * @returns {Promise} Resolving with deviceFeature state.
+ * @example
+ * getState(deviceFeature);
+ */
+async function getState(device, deviceFeature) {
+  const url = getDeviceParam(device, 'CAMERA_URL');
```
Comment (cicoub13) [inline server/services/meross/lib/plug/plug.getState.js]: Why CAMERA_URL ?

Comment (aschelch) [reply server/services/meross/lib/plug/plug.getState.js]: copy/past from rtsp-camera i guess.. 
I'll fix that

**Comment by atrovato** [comment]
Hi, thanks for your PR, I didn't check the code I simply see you didn't test it enough.

But it looks your service automatically add all devices to Gladys. I think you should have a "discover" transition page to list available devices on Meross, and allow to add (or not) each one.

If it's not clear, take a look at eWeLink PR : https://github.com/GladysAssistant/Gladys/pull/719

**Comment by Pierre-Gilles** [comment]
First, thanks for your PR, it's really nice :) 

Completely agree with @atrovato, in Gladys 4 we try to avoid adding devices in the background and prefer a manual approach where the user can cherry pick which device he wants to add to Gladys in the UI.

**With that:** 

- Gladys is not bloated with unwanted devices
- The user can rename/edit the device before adding it to Gladys 
- When there is an error during the configuration process, the error can be displayed in the UI. If it was a background task, the user would probably never see the error and would have to the check the logs, which is not very user-friendly :)

**Comment by aschelch** [comment]
Thanks you for your review, I will work on it this weekend.

**Comment by VonOx** [comment]
Hi @aschelch , I've meross switch so if you need tester ping me.

**Comment by VonOx** [comment]
Hi @aschelch any status on this PR ?

Tell us if you can't commit (no time or whatever ), we can ask on community for a new maintainer

**Pull request closed without merging**